strawberry/arguments.py

```diff
@@ -110,8 +112,8 @@ def convert_argument(value: Any, argument_definition: ArgumentDefinition) -> Any
         kwargs = {}
 
         for field in argument_type._type_definition.fields:
-            if field.name in value:
-                kwargs[field.origin_name] = convert_argument(value[field.name], field)
+            if field.graphql_name in value:
+                kwargs[field.name] = convert_argument(value[field.graphql_name], field)
```

Comment:
Yes probably, although just to be clear `StrawberryField.name` is a property/alias that returns `python_name`

Reply:
Yep understood. It's just good to be consistent internally if we can.

Reply:
Actually, I think I have a bit of work to do here... 
https://github.com/strawberry-graphql/strawberry/blob/94ea99047bdabf85bebc700c807bab4182037a8a/strawberry/field.py#L54-L56

https://github.com/strawberry-graphql/strawberry/blob/94ea99047bdabf85bebc700c807bab4182037a8a/strawberry/field.py#L113-L115

The confusion of `name` strikes again!

Reply:
Ok this should be resolved now. I completely removed `origin_name` (at least from `StrawberryField, it still exists on `ArgumentDefinition` for now), and made sure that the `python_name` and `graphql_name` usages were as expected.

This could probably do with a second set of eyes though.

Reply:
we can do this in another PR if it helps 😊

Reply:
I don't think anything further needs to be changed at this point. I'm just worried that I used the wrong `name` somewhere and our tests aren't robust enough yet to catch it